presto-jdbc/src/main/java/com/facebook/presto/jdbc/PrestoConnection.java

```diff
@@ -642,10 +643,17 @@ private void setCatalogAndSchema()
         }
     }
 
+    @VisibleForTesting
```

Comment:
Why not make the factory method package-private for testing? That seems cleaner since it doesn't depend on state (it's a pure function)